src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -51,21 +55,118 @@
 
 
 /* local function forward declarations */
+static void InsertDistPartitionTuple(Oid distributedRelationId,
+									 text *distributionColumnText,
+									 Oid distributionMethodOid, int colocationId);
 static char LookupDistributionMethod(Oid distributionMethodOid);
 static void RecordDistributedRelationDependencies(Oid distributedRelationId,
 												  Node *distributionKey);
 static Oid SupportFunctionForColumn(Var *partitionColumn, Oid accessMethodId,
 									int16 supportFunctionNumber);
 static bool LocalTableEmpty(Oid tableId);
 static void CreateTruncateTrigger(Oid relationId);
+static int32 LookupColocationId(int shardCount, int replicationFactor,
+								Oid distributionColumnType);
+static int32 GetNextColocationId(void);
+static void InsertDistColocationTuple(int colocationId, int shardCount,
+									  int replicationFactor, Oid distributionColumnType);
+
 
 /* exports for SQL callable functions */
 PG_FUNCTION_INFO_V1(master_create_distributed_table);
+PG_FUNCTION_INFO_V1(create_distributed_table);
 
 
 /*
  * master_create_distributed_table accepts a table, distribution column and
  * method and performs the corresponding catalog changes.
+ */
+Datum
+master_create_distributed_table(PG_FUNCTION_ARGS)
+{
+	Oid distributedRelationId = PG_GETARG_OID(0);
+	text *distributionColumnText = PG_GETARG_TEXT_P(1);
+	Oid distributionMethodOid = PG_GETARG_OID(2);
+
+	InsertDistPartitionTuple(distributedRelationId, distributionColumnText,
+							 distributionMethodOid, INVALID_COLOCATION_ID);
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * create_distributed_table accepts a table, distribution column and
+ * distribution method, then it creates a distributed table.
+ */
+Datum
+create_distributed_table(PG_FUNCTION_ARGS)
+{
+	Oid distributedRelationId = PG_GETARG_OID(0);
+	text *distributionColumnText = PG_GETARG_TEXT_P(1);
+	Oid distributionMethodOid = PG_GETARG_OID(2);
+
+	Relation distributedRelation = NULL;
+	Node *distributionKey = NULL;
+	Var *distributionColumn = NULL;
+	char *distributionColumnName = NULL;
+	int distributionColumnType = 0;
+	int colocationId = 0;
+
+	/* if distribution method is not hash, just create partition metadata */
+	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
+	if (distributionMethod != DISTRIBUTE_BY_HASH)
+	{
+		InsertDistPartitionTuple(distributedRelationId, distributionColumnText,
+								 distributionMethodOid, INVALID_COLOCATION_ID);
+		PG_RETURN_VOID();
+	}
+
+	/* get distribution column type */
+	distributionColumnName = text_to_cstring(distributionColumnText);
+	distributedRelation = relation_open(distributedRelationId, AccessShareLock);
```

Comment:
> I didn't get the final proposal :)

I'm not sure about the lock on pg_dist_partition, that's why I used `approriate ` for now :p